src/main/java/org/elasticsearch/percolator/PercolatorService.java

```diff
@@ -843,15 +847,29 @@ private InternalAggregations reduceAggregations(List<PercolateShardResponse> sha
             return null;
         }
 
+        InternalAggregations aggregations;
         if (shardResults.size() == 1) {
-            return shardResults.get(0).aggregations();
-        }
-
+            aggregations = shardResults.get(0).aggregations();
+        } else {
```

Comment:
Agreed with having min_doc_count=0 on histograms. I think there was some misunderstanding: it is not a new issue but if I read the above code correctly, then we are not going to call reduce() if there is only one shard, which I think is a bug?